python/views/widget_delegate.py

```diff
@@ -114,6 +118,10 @@ def _create_editor_widget(self, model_index, style_options, parent):
         """
         # the default implementation just calls _create_widget for backwards
         # compatibility.
+
+        if not model_index.isValid():
+            return None
```

Comment:
that sounds great - personally, i think it's good to be overly cautios - that's how i tend to code too - recently, i found myself going back to code i had previously written thinking "did i put this in just because i am overly cautious or is it addressing a specific use case?". For that sort of situation, it's really great to have a quick comment to help explain.